From 5765a7aceca01c361748fb70e812a91b4bac52d2 Mon Sep 17 00:00:00 2001
From: Lawrence Rust <lvr@softsystem.co.uk>
Date: Sat, 24 Aug 2013 19:21:42 +0100
Subject: [PATCH 151/333] MediaMon: Don't announce media until after type
 identified

Currently setStatus is called twice for each disk which can cause
2 jumps to handlers.

Signed-off-by: Lawrence Rust <lvr@softsystem.co.uk>
---
 mythtv/libs/libmythbase/mythmedia.cpp |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/mythtv/libs/libmythbase/mythmedia.cpp b/mythtv/libs/libmythbase/mythmedia.cpp
index 7ca6637..c06ee19 100644
--- a/mythtv/libs/libmythbase/mythmedia.cpp
+++ b/mythtv/libs/libmythbase/mythmedia.cpp
@@ -173,8 +173,7 @@ bool MythMediaDevice::performMountCmd(bool DoMount)
                                            " find mounted media, but failed?");
                     return false;
                 }
-                setStatus(MEDIASTAT_MOUNTED); // emit statusChanged
-                onDeviceMounted();
+                onDeviceMounted(); // Identify disk type & content
                 LOG(VB_GENERAL, LOG_INFO,
                         QString("Detected MediaType ") + MediaTypeString());
             }
-- 
1.7.9.5

